Trying to use (unsupported) equality constraint in component makes leptos ignore the component and its errors and report success

Hi, thanks a lot for creating/maintaining this leptos and happy holidays!

**Describe the bug**
Trying to use (unsupported) equality constraint makes leptos ignore the component and its errors and report compilation success

**Leptos Dependencies**

For example:
```toml
leptos = "0.5.4"
```

**To Reproduce**
Minimum reproducible example:

```rust
use leptos::*;

pub fn main() {
    mount_to_body(|| view! {
        // <BugExample />
    })
}

#[component]
pub fn BugExample<T>() -> impl IntoView
where
    T = String,
{
    is_let_even_required? = [1, 2];
    rust_looks_like_python_nowadays.get(3, default="abc");
    std::fs::File
}
```

1. Run `cargo check --target wasm32-unknown-unknown`
```console
$ cargo check --target wasm32-unknown-unknown
    Checking leptos-no-error v0.1.0 (/home/bilko/git/leptos-no-error)
    Finished dev [unoptimized + debuginfo] target(s) in 0.13s
$ echo $?
0
```

2. Run `cargo expand`
```console
$ cargo expand
    Checking leptos-no-error v0.1.0 (/home/bilko/git/leptos-no-error)
    Finished dev [unoptimized + debuginfo] target(s) in 0.14s

#![feature(prelude_import)]
#[prelude_import]
use std::prelude::rust_2021::*;
#[macro_use]
extern crate std;
use leptos::*;
pub fn main() {
    mount_to_body(|| { leptos::leptos_dom::Unit })
}
```

3. Uncomment `// <BugExample />` and run again
```console
$ cargo check --target wasm32-unknown-unknown
    Checking leptos-no-error v0.1.0 (/home/bilko/git/leptos-no-error)
error[E0425]: cannot find value `BugExample` in this scope
 --> src/main.rs:5:10
  |
5 |         <BugExample />
  |          ^^^^^^^^^^ not found in this scope

For more information about this error, try `rustc --explain E0425`.
error: could not compile `leptos-no-error` (bin "leptos-no-error") due to previous error
```

4. Remove or comment out `#[component]` and see all errors:
```console
$ cargo check --target wasm32-unknown-unknown
    Checking leptos-no-error v0.1.0 (/home/bilko/git/leptos-no-error)
error: equality constraints are not yet supported in `where` clauses
  --> src/main.rs:12:5
   |
12 |     T = String,
   |     ^^^^^^^^^^ not supported
   |
   = note: see issue #20041 <https://github.com/rust-lang/rust/issues/20041> for more information

error[E0425]: cannot find value `is_let_even_required` in this scope
  --> src/main.rs:14:5
   |
14 |     is_let_even_required? = [1, 2];
   |     ^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `rust_looks_like_python_nowadays` in this scope
  --> src/main.rs:15:5
   |
15 |     rust_looks_like_python_nowadays.get(3, default="abc");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `default` in this scope
  --> src/main.rs:15:44
   |
15 |     rust_looks_like_python_nowadays.get(3, default="abc");
   |                                            ^^^^^^^
   |
help: you might have meant to introduce a new binding
   |
15 |     rust_looks_like_python_nowadays.get(3, let default="abc");
   |                                            +++

error[E0423]: expected value, found struct `std::fs::File`
   --> src/main.rs:16:5
    |
16  |     std::fs::File
    |     ^^^^^^^^^^^^^ help: use struct literal syntax instead: `std::fs::File { inner: val }`
    |
   ::: /home/bilko/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/fs.rs:100:1
    |
100 | pub struct File {
    | --------------- `std::fs::File` defined here

error[E0070]: invalid left-hand side of assignment
  --> src/main.rs:14:27
   |
14 |     is_let_even_required? = [1, 2];
   |     --------------------- ^
   |     |
   |     cannot assign to this expression

Some errors have detailed explanations: E0070, E0423, E0425.
For more information about an error, try `rustc --explain E0070`.
error: could not compile `leptos-no-error` (bin "leptos-no-error") due to 6 previous errors
```

5. Uncomment `#[component]`, comment the full where constraint and see all errors again (errors ommited, but same as above).

**Expected behavior**
On step 1. and 3., all errors should be reported. On step 1., compilation should not be successful.

**Additional context**
Although this does not happen for valid programs, it can be confusing when developing as you need to comment `#[component]` out to be able to debug it.

--- a/leptos_macro/src/lib.rs
+++ b/leptos_macro/src/lib.rs
@@ -603,7 +603,7 @@ pub fn component(args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
     };
 
     let Ok(mut dummy) = syn::parse::<DummyModel>(s.clone()) else {
-        return s.into();
+        return s;
     };
     let parse_result = syn::parse::<component::Model>(s);
 
@@ -703,11 +703,11 @@ pub fn component(args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
 #[proc_macro_attribute]
 pub fn island(_args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
     let Ok(mut dummy) = syn::parse::<DummyModel>(s.clone()) else {
-        return s.into();
+        return s;
     };
     let parse_result = syn::parse::<component::Model>(s);
 
-    if let (Ok(ref mut unexpanded), Ok(model)) = (&mut dummy, parse_result) {
+    if let (ref mut unexpanded, Ok(model)) = (&mut dummy, parse_result) {
         let expanded = model.is_island().into_token_stream();
         if !matches!(unexpanded.vis, Visibility::Public(_)) {
             unexpanded.vis = Visibility::Public(Pub {
@@ -722,15 +722,13 @@ pub fn island(_args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
             #[allow(non_snake_case, dead_code, clippy::too_many_arguments)]
             #unexpanded
         }
-    } else if let Ok(mut dummy) = dummy {
+    } else {
         dummy.sig.ident = unmodified_fn_name_from_fn_name(&dummy.sig.ident);
         quote! {
             #[doc(hidden)]
             #[allow(non_snake_case, dead_code, clippy::too_many_arguments)]
             #dummy
         }
-    } else {
-        s.into()
     }
     .into()
 }

--- a/leptos_macro/src/lib.rs
+++ b/leptos_macro/src/lib.rs
@@ -602,10 +602,12 @@ pub fn component(args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
         false
     };
 
-    let mut dummy = syn::parse::<DummyModel>(s.clone());
-    let parse_result = syn::parse::<component::Model>(s.clone());
+    let Ok(mut dummy) = syn::parse::<DummyModel>(s.clone()) else {
+        return s.into();
+    };
+    let parse_result = syn::parse::<component::Model>(s);
 
-    if let (Ok(ref mut unexpanded), Ok(model)) = (&mut dummy, parse_result) {
+    if let (ref mut unexpanded, Ok(model)) = (&mut dummy, parse_result) {
         let expanded = model.is_transparent(is_transparent).into_token_stream();
         unexpanded.sig.ident =
             unmodified_fn_name_from_fn_name(&unexpanded.sig.ident);
@@ -615,15 +617,13 @@ pub fn component(args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
             #[allow(non_snake_case, dead_code, clippy::too_many_arguments)]
             #unexpanded
         }
-    } else if let Ok(mut dummy) = dummy {
+    } else {
         dummy.sig.ident = unmodified_fn_name_from_fn_name(&dummy.sig.ident);
         quote! {
             #[doc(hidden)]
             #[allow(non_snake_case, dead_code, clippy::too_many_arguments)]
             #dummy
         }
-    } else {
-        s.into()
     }
     .into()
 }
@@ -702,8 +702,10 @@ pub fn component(args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
 #[proc_macro_error::proc_macro_error]
 #[proc_macro_attribute]
 pub fn island(_args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
-    let mut dummy = syn::parse::<DummyModel>(s.clone());
-    let parse_result = syn::parse::<component::Model>(s.clone());
+    let Ok(mut dummy) = syn::parse::<DummyModel>(s.clone()) else {
+        return s.into();
+    };
+    let parse_result = syn::parse::<component::Model>(s);
 
     if let (Ok(ref mut unexpanded), Ok(model)) = (&mut dummy, parse_result) {
         let expanded = model.is_island().into_token_stream();

--- a/leptos_macro/src/lib.rs
+++ b/leptos_macro/src/lib.rs
@@ -603,7 +603,7 @@ pub fn component(args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
     };
 
     let mut dummy = syn::parse::<DummyModel>(s.clone());
-    let parse_result = syn::parse::<component::Model>(s);
+    let parse_result = syn::parse::<component::Model>(s.clone());
 
     if let (Ok(ref mut unexpanded), Ok(model)) = (&mut dummy, parse_result) {
         let expanded = model.is_transparent(is_transparent).into_token_stream();
@@ -623,7 +623,7 @@ pub fn component(args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
             #dummy
         }
     } else {
-        quote! {}
+        s.into()
     }
     .into()
 }
@@ -703,7 +703,7 @@ pub fn component(args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
 #[proc_macro_attribute]
 pub fn island(_args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
     let mut dummy = syn::parse::<DummyModel>(s.clone());
-    let parse_result = syn::parse::<component::Model>(s);
+    let parse_result = syn::parse::<component::Model>(s.clone());
 
     if let (Ok(ref mut unexpanded), Ok(model)) = (&mut dummy, parse_result) {
         let expanded = model.is_island().into_token_stream();
@@ -728,7 +728,7 @@ pub fn island(_args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
             #dummy
         }
     } else {
-        quote! {}
+        s.into()
     }
     .into()
 }

